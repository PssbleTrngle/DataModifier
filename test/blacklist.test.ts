import setupLoader from './shared/loaderSetup.js'
import createTestAcceptor from './mock/TestAcceptor.js'

const { loader } = setupLoader()

describe('blacklist tests', () => {
   it('generated a jei blacklist config file', async () => {
      const acceptor = createTestAcceptor()

      loader.blacklist.hide('minecraft:stone')
      loader.blacklist.hide({ fluid: 'water' })
      loader.blacklist.hide({ block: 'water' })
      loader.blacklist.hide(['ice', { fluid: 'forge:milk' }])

      await loader.emit(acceptor)

      expect(acceptor.at('jei/blacklist.cfg')).toMatchSnapshot('jei blacklist config file')
   })
   it('does not create the jei blacklist config if nothing is hidden', async () => {
      const acceptor = createTestAcceptor()

      await loader.emit(acceptor)

      expect(acceptor.at('jei/blacklist.cfg')).toBeNull()
   })
})
