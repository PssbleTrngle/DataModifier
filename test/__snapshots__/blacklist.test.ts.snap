// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`blacklist tests generated a jei blacklist config file: jei blacklist config file 1`] = `
"minecraft:stone
minecraft:water
minecraft:ice
forge:milk"
`;
